src/main/java/duke/command/ListCommand.java

```diff
@@ -0,0 +1,51 @@
+package duke.command;
+
+import duke.*;
+import duke.datetime.DateTimeUtility;
+import duke.storage.Storage;
+import duke.task.TaskList;
+import duke.ui.Ui;
+
+public class ListCommand extends Command {
+    private String by;
+
+    public ListCommand() {
+        super();
+        this.by = "";
+        this.cmd = CMD.LIST;
+    }
+
+    public ListCommand(String by) {
+        this();
+        this.by = DateTimeUtility.formatString(by);
+    }
+
+    @Override
+    public void execute(TaskList taskList, Ui ui, Storage storage) throws DukeException {
```

Comment:
Do you think the nesting may be a bit deep here? I tend to find nested if-else hard to read when it goes down to 3-4 layers. One possible way of avoiding this:

```
if (taskList.isEmpty()) {
    ui.display("UR LIST HAZ NUTHIN LOLOL");
    return; // Stop the rest of the method from executing
}

if (by.isEmpty()) {
/* continue from here */
```